core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -369,6 +369,7 @@ public void close() throws IOException {
       ScheduledExecutorService service = refreshExecutor;
       this.refreshExecutor = null;
 
+      service.shutdown();
```

Comment:
Good catch.

One question: do we potentially want to do this inside of a try block? [Looking at the javadocs](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html#shutdown--), it seems like the only exception it can throw is `SecurityException` which likely shouldn't happen if the executor service was started.

Reply:
My thought is that a try/catch block might be overkill, but I can add that if desired

Reply:
Agreed. This is sufficient for now in my opinion. I also think the example from the javadoc might be overkill (if anything the results will simply be processed and not returned to anything by the server, which happens all the time during shutdown etc).

Reply:
I'm thinking it is better to throw, so if the `shudown()` fails, it won't proceed to the `awaitTermination()`